src/Features/Protocol/LanguageServices/RoslynLanguageService.cs

```diff
@@ -0,0 +1,673 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.DocumentHighlighting;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Editor.Extensibility.NavigationBar;
+using Microsoft.CodeAnalysis.Editor.FindUsages;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.NavigateTo;
+using Microsoft.CodeAnalysis.Navigation;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Protocol.LanguageServices.Extensions;
+using Microsoft.CodeAnalysis.QuickInfo;
+using Microsoft.CodeAnalysis.Structure;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using VSLocation = Microsoft.VisualStudio.LanguageServer.Protocol.Location;
+
+namespace Microsoft.CodeAnalysis.Protocol.LanguageServices
+{
+    /// <summary>
+    /// This is the roslyn language service.
+    ///
+    /// This glues the LSP requests and solution to 
+    /// Roslyn (and services) and outputs LSP types.
+    /// </summary>
+    public sealed class RoslynLanguageService
```

Comment:
>public sealed class RoslynLanguageService [](start = 4, length = 41)

Should not be public.

Reply:
My understanding was these were to be vended as a nuget package to be publicly consumed @heejaechang

Reply:
@dibarbet for now, leave it to internal since we will address LiveShare case first.